File: BE/src/main/java/com/react/auth/util/ResponseUtil.java
package com.react.auth.util;

import jakarta.servlet.http.HttpServletResponse;

import java.io.IOException;
import java.io.PrintWriter;

public final class ResponseUtil {

    private ResponseUtil() {
        // prevent instantiation
    }

    public static void writeError(
            HttpServletResponse response,
            int status,
            String message
    ) throws IOException {

        response.setStatus(status);
        response.setContentType("application/json");
        response.setCharacterEncoding("UTF-8");

        try (PrintWriter out = response.getWriter()) {
            out.write("{\"message\":\"" + message + "\"}");
        }
    }
}

File: BaseSpringSecurity/src/main/kotlin/utils/ResponseUtil.kt
package com.utils
import jakarta.servlet.http.HttpServletResponse

object ResponseUtil {
    fun writeError(response: HttpServletResponse, status: Int, message: String) {
        response.status = status
        response.contentType = "application/json"
        response.characterEncoding = "UTF-8"
        response.writer.use { out ->
            out.write("""{"message":"$message"}""")
        }
    }
}